standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PartitionManagementTask.java

```diff
@@ -169,7 +168,7 @@ public void run() {
           // this always runs in 'sync' mode where partitions can be added and dropped
           MsckInfo msckInfo = new MsckInfo(table.getCatName(), table.getDbName(), table.getTableName(),
             null, null, true, true, true, retentionSeconds);
-          executorService.submit(new MsckThread(msckInfo, msckConf, qualifiedTableName, countDownLatch));
+          executorService.submit(new MsckThread(msckInfo, msckConf, qualifiedTableName, countDownLatch, msc));
```

Comment:
Sorry, looks like I missed to notice this change earlier. Sharing HMS client across threads might not be good idea. If all threads makes the call at the same time, the results might get mixed up. It would be better to use separate client for each of them.